tests/test_dynamodb2/test_dynamodb.py

```diff
@@ -6008,3 +6008,302 @@ def test_get_item_for_non_existent_table_raises_error():
+    restored = client.restore_table_from_backup(
+        TargetTableName=restored_table_name, BackupArn=backup_arn
+    ).get("TableDescription")
+    restored.should.have.key("AttributeDefinitions").should.equal(
+        table["AttributeDefinitions"]
+    )
+    restored.should.have.key("TableName").should.equal(restored_table_name)
+    restored.should.have.key("KeySchema").should.equal(table["KeySchema"])
+    restored.should.have.key("TableStatus")
+    restored.should.have.key("ItemCount").should.equal(5)
+    restored.should.have.key("TableArn").should.contain(restored_table_name)
+    restored.should.have.key("RestoreSummary").should.be.a(dict)
+    summary = restored.get("RestoreSummary")
+    summary.should.have.key("SourceBackupArn").should.equal(backup_arn)
+    summary.should.have.key("SourceTableArn").should.equal(table["TableArn"])
+    summary.should.have.key("RestoreDateTime").should.be.a(datetime)
+    summary.should.have.key("RestoreInProgress").should.equal(False)
+
+
+@mock_dynamodb2
+def test_delete_non_existent_backup_raises_error():
+    client = boto3.client("dynamodb", "us-east-1")
+    non_existent_arn = "arn:aws:dynamodb:us-east-1:123456789012:table/table-name/backup/01623095754481-2cfcd6f9"
+    with pytest.raises(ClientError) as ex:
+        client.delete_backup(BackupArn=non_existent_arn)
+    error = ex.value.response["Error"]
+    error["Code"].should.equal("BackupNotFoundException")
+    error["Message"].should.equal("Backup not found: {}".format(non_existent_arn))
+
+
+@mock_dynamodb2
+def test_delete_backup():
+    client = boto3.client("dynamodb", "us-east-1")
+    table_name = "test-table-1"
+    backup_names = ["backup-1", "backup-2"]
+    client.create_table(
+        TableName=table_name,
+        KeySchema=[{"AttributeName": "id", "KeyType": "HASH"}],
+        AttributeDefinitions=[{"AttributeName": "id", "AttributeType": "S"}],
+        ProvisionedThroughput={"ReadCapacityUnits": 5, "WriteCapacityUnits": 5},
+    )
+    for backup_name in backup_names:
+        client.create_backup(TableName=table_name, BackupName=backup_name)
+    resp = client.list_backups(TableName=table_name, BackupType="USER")
+    resp["BackupSummaries"].should.have.length_of(2)
+    backup_to_delete = resp["BackupSummaries"][0]["BackupArn"]
+    backup_deleted = client.delete_backup(BackupArn=backup_to_delete).get(
+        "BackupDescription"
+    )
+    backup_deleted.should.have.key("SourceTableDetails")
+    backup_deleted.should.have.key("BackupDetails")
+    details = backup_deleted["BackupDetails"]
+    details.should.have.key("BackupArn").should.equal(backup_to_delete)
+    details.should.have.key("BackupName").should.be.within(backup_names)
+    details.should.have.key("BackupStatus").should.equal("DELETED")
+    resp = client.list_backups(TableName=table_name, BackupType="USER")
+    resp["BackupSummaries"].should.have.length_of(1)
+
+
+@mock_dynamodb2
+def test_source_and_restored_table_items_are_not_linked():
```

Comment:
Can we extend this test a bit, to verify the behaviour for items created between backup-creation and restoring?
```
create_backup
add_items to original table
restore_backup
verify original has x items
verify backup only has y items
```